dipy/viz/tests/test_ui.py

```diff
@@ -627,6 +627,78 @@ def test_ui_image_container_2d(interactive=False):
         show_manager.start()
 
 
+@npt.dec.skipif(not have_vtk or skip_it)
+@xvfb_it
+def test_ui_file_menu_2d(interactive=False):
+    filename = "test_ui_file_menu_2d"
+    recording_filename = pjoin(DATA_DIR, filename + ".log.gz")
+    expected_events_counts_filename = pjoin(DATA_DIR, filename + ".pkl")
+
+    # Create temporary directory and files
+    os.mkdir(os.path.join(os.getcwd(), "testdir"))
+    os.chdir("testdir")
+    os.mkdir(os.path.join(os.getcwd(), "tempdir"))
+    for i in range(10):
+        open(os.path.join(os.getcwd(), "tempdir", "test" + str(i) + ".txt"),
+             'wt').close()
+    open("testfile.txt", 'wt').close()
+
+    filemenu = ui.FileMenu2D(size=(500, 500), extensions=["txt"],
+                             directory_path=os.getcwd())
+
+    # We will collect the sequence of files that have been selected.
+    selected_files = []
+
+    def _on_change():
+        selected_files.append(list(filemenu.listbox.selected))
+
+    # Set up a callback when selection changes.
+    filemenu.listbox.on_change = _on_change
+
+    # Assign the counter callback to every possible event.
+    event_counter = EventCounter()
+    event_counter.monitor(filemenu)
+
+    # Create a show manager and record/play events.
+    show_manager = window.ShowManager(size=(600, 600),
+                                      title="DIPY FileMenu")
+    show_manager.ren.add(filemenu)
+
+    # Recorded events:
+    #  1. Click on 'testfile.txt'
+    #  2. Click on 'tempdir/'
+    #  3. Click on 'test0.txt'.
+    #  4. Shift + Click on 'test6.txt'.
+    #  5. Click on '../'.
+    #  2. Click on 'testfile.txt'.
+    show_manager.play_events_from_file(recording_filename)
```

Comment:
Where does the recording file come from? I mean how do you record it? I don't see any `EventCounter.save`.

Reply:
The events are already recorded and saved. I removed the recording part because in the earlier implementation, there was no `interactive` parameter and if a user wanted to interact with the elements, they would have to set `recording = True` which changes the log files.
Maybe we can have two parameters `recording` and `interactive`?

Reply:
That's a good idea. We could have a single parameter `mode=None|'interactive|'recording'`. I found the recording useful when we change some core functionality of the UI framework and have to redo the recording. But to be fair that happens less often nowadays.

Reply:
I'll do that in another PR since all the tests will have to be changed.
@dmreagan @Garyfallidis @ranveeraggarwal should I go ahead with this?